### PR TITLE
Install Java from Adoptium

### LINKED ISSUE #232: Update Java role to use Eclipse Temurin Java releases, so we can install e.g. Java 21 on CentOS 7 or Java 8 on RHEL 10

### PR DESCRIPTION
**Pull request by p-j-smith**
Fixes #232 

- update the `install_java` role to use the Adoptium distribution (Eclipse Temurin)
- the default java version is kept the same (8)
- remove `java_package` variable and replace it with `java_major_version`
- set the default JRE directory (using `community.general.alternatives`). openjdk does this automatically, but we need to do it manually for Temurin
- change the keystore path to `/etc/pki/java/cacerts` (which is provided by `ca-certificates`). The previous path was actually a symlink to this one anyway

### CODE REVIEW

File: roles/install_java/defaults/main.yml [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+java_package: temurin-{{ java_major_version }}-jdk
 java_home: /usr/lib/jvm/jre
-java_package: java-1.8.0-openjdk
-java_package_version: devel
+
+java_adoptium_repo_baseurl:
+  https://packages.adoptium.net/artifactory/rpm/rhel/{{
+  ansible_facts['distribution_major_version'] }}/{{
+  ansible_facts['architecture'] }}
+java_adoptium_repo_gpgkey: https://packages.adoptium.net/artifactory/api/gpg/key/public
```
Comment (HChughtai) [inline roles/install_java/defaults/main.yml:12]: Do you think there's any benefit to making this even more generic so we can use a different distribution if we wished to?

**Review by HChughtai — APPROVED** [review]
Looks great, thanks for figuring this out and getting it in! A nice step on the way to XNAT 1.10

My only comments are whether we should at this stage think about making the choice of Java distribution more generic.

File: roles/install_java/defaults/main.yml (hunk 1 above)
Comment (p-j-smith) [reply roles/install_java/defaults/main.yml:12]: Yeah it might be worthwhile. Although I think it could get quite complicated if we want to support any distribution. Plus we'd want to test each one that we might want to use, just to check for edge cases. I can open an issue for it, but might not add it in this PR if that's okay?

**Comment by p-j-smith** [comment]
I've messed up my PR stack, so I'll merge this into `main` and close the other 3 PRs

File: roles/install_java/defaults/main.yml (hunk 1 above)
Comment (HChughtai) [reply roles/install_java/defaults/main.yml:12]: That's fair, and I don't expect it to be a need anytime soon so can leave it until/if it's a need
